### ISSUE TITLE
Custom Fields OR Add-Ons with a fee associated

### ISSUE THREAD

**Issue by geoffjohnsonphoto** [calcom/cal.com#7075]
### Is your proposal related to a problem?

All of my competitors are using Acuity software with a great feature of being able to have add-on's with fees associate to them.
For example, my base rental is $170/ 2 hours, but if they choose a specific piece of equipment to rent, I want to be able to charge more for that.

### Describe the solution you'd like

A simple solution would be to have this connected in the custom field area.  
### Describe alternatives you've considered

Nothing really.  Other than tell the customer to send me an email to get a quote.

### Additional context

[(Write your answer here.)
](https://cal.com/lumierestudio/book?type=179415&date=2023-02-28T09%3A45%3A00-06%3A00&slug=2-hours-studio-rental&count=)

**Comment by PeerRich** [comment]
we are reworking the payment provider right now and could think about addons. interesting idea @zomars 

definitely needs prioritization and needs design

**Comment by PeerRich** [comment]
@Jaibles this could be a check box per addon on the booking page with a price next to it

**Comment by Hybes** [comment]
@PeerRich can this be revisited and possibly re-prioritised?

This is in all competitors and should probably be looked at with a slightly higher priority.

**Comment by Hybes** [comment]
Maybe bounty it?

**Comment by PeerRich** [comment]
hmm so far we havent gotten many requests 

why not one event type per product?

**Comment by geoffjohnsonphoto** [comment]
I would absolutely switch to use Cal as it’s much cleaner and I do love the UI if we can solve multiple add ons options during checkout. 

Here’s what I can currently do in Acuity. 

![IMG_9842](https://github.com/calcom/cal.com/assets/125304204/c520c7e9-6ff4-499d-b37e-5aa31e2bfc28)

**Comment by PeerRich** [comment]
can you also post how the setup looks like?

**Comment by Hybes** [comment]
@geoffjohnsonphoto yes, this exactly! just the ability to add them as addons, increasing the total price, adds a large amount of possibility with one tweak.

**Comment by Hybes** [comment]
@geoffjohnsonphoto can you post the acuity setup page for these addons, so we can hopefully get some movement on this?

**Comment by Mihir867** [comment]
Hi, I would like to try this one, Can I ?

**Comment by Hybes** [comment]
any movement on this?

**Comment by Hybes** [comment]
@Mihir867 i dont see why you can't go for an attempt?

@zavredevs did you make a pr for this?

**Comment by Aspireve** [comment]
Hey Hi!!
Could you explain it a bit more

**Comment by Mihir867** [comment]
Hi, I was working to add it, Just had a quick question if I should add it in the Events Tab> Advanced settings > Offer Seats like this 
![image](https://github.com/user-attachments/assets/fa496dea-4d90-4e4e-915b-5b3a4848f75b)
or should I add it in Apps section like here - 
![WhatsApp Image 2024-08-06 at 5 48 44 PM](https://github.com/user-attachments/assets/837fbca7-bad9-493b-8888-e4044fa81e27)

**Comment by Mihir867** [comment]
@ciaranha  Can we add it on the advanced settings page ?

**Comment by Hybes** [comment]
Would this not essentially have it's own tab, alongside app, advanced, etc.

Called Addons, it would need to support the payment apps and bolt-on extra costs to it, should those new items be selected/added on the booking page.

**Comment by Mihir867** [comment]
Oh sure, will create a new tab, Do I need a design to get started ?

**Comment by Hybes** [comment]
> Oh sure, will create a new tab, Do I need a design to get started ?

one for @zomars or @PeerRich

**Comment by Hybes** [comment]
@PeerRich @zomars can we start getting this one moving again?

**Comment by zomars** [comment]
Hey @Hybes we're currently focused on other priorities. As always contributions are always welcome. Just me mindful of our contribution guidelines. 🙏

**Comment by Hybes** [comment]
> Hey @Hybes we're currently focused on other priorities. As always contributions are always welcome. Just me mindful of our contribution guidelines. 🙏

Yeah that's all good, just that Mihir867 had wanted to know some answers in order to get this moving.

**Comment by Hybes** [comment]
I feel its' logical to be an app or its own tab in the sections realistically.

**Comment by serunkuma** [comment]
/attempt [#7075](https://github.com/calcom/cal.com/issues/7075)

**Comment by PeerRich** [comment]
added some designs:

if a payment app is enabled, we could add a price input next to each booking question:

Select:
![Image](https://github.com/user-attachments/assets/e452f3f7-690a-4417-bc6c-faf8a3490bef)

Checkbox:

![Image](https://github.com/user-attachments/assets/8c967116-dedd-4533-a5a8-ee799bc272f7)

Number: (a number input would multiply per number, so an input of 20 and price per unit of $10 would make a $200 addon)

![Image](https://github.com/user-attachments/assets/72939fa0-b5fa-4b19-958d-2aff10262540)

**Comment by PeerRich** [comment]
this addon feature should work for all paid apps

**Comment by jevil25** [comment]
Hey @PeerRich I can take this up

**Comment by jevil25** [comment]
can you explain more on how  can access this dialog box

**Comment by vineetk13** [comment]
@PeerRich WIth a payment app enabled, how can a user add Select, Checkbox or Number input **without** the pricing options/labels while creating a new booking question?

**Comment by jayantpranjal0** [comment]
@CarinaWolli looks like #19997 is not being worked upon. Is it okay if I pick it up from there?

**Comment by CarinaWolli** [comment]
> @CarinaWolli looks like https://github.com/calcom/cal.com/pull/19997 is not being worked upon. Is it okay if I pick it up from there?


For sure, feel free to open a PR for it. The better attempt will be merged 🙏
